### DISCUSSION TITLE
Updating/Installing Astroid 3 for Joomla 4 & 5 Guidelines

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#487] [Q&A] [answered]
Hello

I want to make more video guides and this seems as good a place as any…
Starting first with a fresh install of Astroid on Joomla 4.
Will get to Joomla 5 after making sure that this is sound…

**Fresh Install on Joomla 4**

- Make sure that correct Joomla version is installed 4.4.0 min
- if stuck on 4.3.4 or earlier for what ever reason follow steps below but with latest:
astroid-package-v2.x.x.Framework-Template.zip
 
- back up: Akeeba/Cpanel Backup or similar
- install the latest version of Astroid:
astroid-package-v3.x.x.Framework-Template.zip

- use the standard Joomla install from file method to install the Astroid package
- make sure that Astroid is active in the Joomla Template Section
- check front end
- navigate to back end and make changes…

**Update Astroid from 2 to 3**

- Make sure that correct Joomla version is installed 4.4.0 min
- if stuck on 4.3.4 or earlier for what ever reason follow steps below but with latest:
astroid-framework-2.x.x.zip
 
- back up: Akeeba/Cpanel Backup or similar
- use the export feature in the Astroid back end to make a back up of any template changes made
- install the latest version of Astroid:
astroid-framework-3.x.x.zip

- use the standard Joomla install from file method to install the Framework package
- make sure that Astroid is active in the Joomla Template section
- check front end
- navigate to back end and make changes…

Am I missing something/Any Gotchas?

thanks in advance

Paul

**Comment by sonvnn** [comment]
Hi Paul,

Thank you detail! It is great. I think it's quite complete and you can start writing now :)

Best Regards,
Sonny

**Comment by paurray** [comment]
Still collecting ideas but I think think I will call this one “Astroid under the Hood” or similar.

- What I have learnt and will also mention is that as long as something works on Joomla 4.4 it should work on Joomla 5. So need to go into extra detail here. All bang up to date on live site with no problems.

- I think mentioning that the Import/Export options have moved would make sense:

How to reuse template on other sites: https://github.com/templaza/astroid-framework/discussions/490

- Maybe touching on this eg the scenario when a new template is released in a package

Need a hand for Astroid 3.0.3 Testing: https://github.com/templaza/astroid-framework/discussions/425

“I am working on Two template. I will update One and Zero after that. I am working to rebuild everything of Astroid. It's an envolution. At the moment, you can still use B/C to work with Zero. I will release Two in the end of this month.”

@sonvnn 
This means (assuming that I already have Astroid installed) it is time to reinstall the full package AGAIN if access to the new template is required? Right?

Also wondering if I export a template that i have built using Zero or One as a basis. Into eg template Two will I be able to do this and get more speed? I guess yes?

thanks

Paul

**Comment by sonvnn** [comment]
@paurray I am sorry for late reply! Just busy with a project of my company.

I am still working on Two. I hope I can give something for you to testing in the next week. It's a new template of Astroid and separate with One and Zero. I will release a quickstart package with Two, so you can easy to test all Astroid functions. 

I updated the Import/Export document https://docs.astroidframe.work/more/export-import You can check it.

Please let me know if you have further question. I am busy right now, but I will try to answer one by one later :)

Best Regards,
Sonny

**Comment by paurray** [comment]
@sonvnn thanks for getting back to me.
Yes great that you have updated the documentation

I started on a script for a guide for beginners it is a work in progress/about 70% there:

01) Installing Astroid but where to start?
02) Brief overview of current Joomla Versions
03) Brief overview of current Astroid Versions
04) Joomla 3.3.4 or less & latest Astroid 2 Version
05) Joomla 3.4 or higher & latest Astroid 3 Version
06) When to use the Full Package including Templates/or just Frame work
07) Current State of Play with respect to Astroid Templates (referring to “Two”)
08) Having got Astroid installed what about updating it
09) What to do if a new Template comes out (referring to “Two”)
10) Updating Joomla and Astroid parallel eg going Astroid 2 - 3 whilst going Joomla 4 -5!
11) General Tips when updating Anything
12) Wrap Up

I am on holidays from the 18th Dec through to first Week of Jan so will not get it finished this year, will also not be able to test anything this year :-( but happy to next year.

In the mean time could you have a look at this when ever you have a chance:

Easier to understand onboarding for new Astroid Users https://github.com/templaza/astroid-framework/discussions/506

If anything in there makes sense maybe it could be implemented by you and I could add changes made to a "Updating/Installing Astroid 3 for Joomla 4 & 5 Guidelines" video early next year? 

What do you say?

all the best

Paul

**Comment by Chacapamac** [comment]
@paurray 
**All that look good my friend!**

I think an important point that can be complicate is the transition from Joomla 4.1 and Astroid 2.6.0 UP, where Joomla introduce the new “Child Templates”. Maybe it can be a tutorial on its own?

I have some infos that I already prepare here that probably can be use.
The following is a Mac RTF file

[Astroid-Child Templates.zip](https://github.com/templaza/astroid-framework/files/13686088/Astroid-Child.Templates.zip)

**Comment by paurray** [comment]
Hi @Chacapamac 
It is funny before skimming through your amazing documentation I watched the exact video you linked in there:
Child Templates in Joomla 4.1
https://www.youtube.com/watch?v=_3cUTEUvczk

I am curious do you happen to know off the top of your head if any of the other providers such as Joomshaper or Yootheme are onboard with the native Joomla Child Template thing now in almost 2024? If not it seems to be just yet another huge reason to use Astroid over other solutions… I completely get what you are getting at.

I guess what would then make sense is:

Updating Astroid 2 beyond Joomla 4.1 as in from Joomla 4.0+
and
Updating Astroid 3 beyond Joomla 4.4 and beyond including Joomla 5

???

Don't want to make promises I can not keep but I will look into this in more detail at the start of 2024
The whole Child Template thing is very much currently on the outer edge of my current skill set. But never say never.
Maybe if I have enough time and if you can review video(s) tutorial we can do this together?

thanks as always
Paul

**Reply by Chacapamac** [reply]
@paurray 
To my knowledge, none of the big frameworks _(Other than Astroid)_ are compatible with the **Native Joomla Child Template System**. I think they only offer some hybrid proprietary systems, making Astroid, again, the best Framework Companion to Joomla. I hate all the frameworks that try to reinvent and impose their way over the unbeatable Joomla system. I love that Astroid is branching and relying 100% on the awesomeness power of Joomla.

Good Day my Friend!

**Comment by paurray** [comment]
@Chacapamac 
Well for what it is worth in MAMP I have:

Astroid Framework Version: 2.5.20
Template Version: 2.5.19
Working in Joomla 4.0.6
Running on PHP 8.1.10

All fresh no changes made to date.
I have cloned the site and made a snapshot!

So at least I can technically replicated PRE Joomla 4.1 & Astroid 2.6 state and can go back to test multiple times...

Off for xmas break now back in the new year…

**Reply by Chacapamac** [reply]
Cool @paurray  
**Hope the best for your family in the New Year approaching!**

**Comment by paurray** [comment]
@Chacapamac @sonvnn 
_Draft Blog Post is looking something like this (any feedback welcome) there is a related video soon going into more detail in the works:_

**First Time Install of the Astroid Frame Work for Joomla**

Are you new to the Astroid Framework? Maybe even a little intimidated by Git Hub? The multiple Astroid releases with multiple file types and often seemingly cryptic file names? You want to get get cracking but just don't know where to start? Because when push comes to shove the Astroid install is in and of itself is pretty straight forward but there are a couple of things to look out for before getting up and running.

Because Joomla is constantly being improved from time to time mission critical items get altered at the Joomla end. I am not going to pretend to have a thorough understanding of this but at the most basic level sometimes folders and files get modified and shunted around the place. The Astroid update cycle is always slightly behind the Joomla update cycle. Joomla being the horse and Astroid being the carriage. The Astroid version you install has links to hundreds if not thousands of Joomla folders and files and you want to avoid a potential Joomla/Astroid release version mismatch.

If you keep your site bang up to date all of the time this is not a big deal. Then all you need to do is install a recent version of the Astroid Frame Work (INSERT LINK HERE) and you should be good to go. Just make sure that the release version of Astroid that you are installing is released after the latest version of Joomla and double check the Astroid Release notes for hints and clues. Go back a couple of versions and double check the notes again. If you really want to err on the side of caution you can always hold off on a fresh brand new up to date install for a week or two. Whilst keeping an eye on the Astroid discussions forum (INSERT LINK HERE) for any know issues with the latest and greatest releases. As with any update you should make a back up before updating.

But what If your Joomla version is not bang up to date. And assuming that you are on Joomla 4 or higher? If you are on pre Joomla 4 then maybe bookmark this and come back to it if and when you are ready. Also bear in mind if it works on Joomla 4.4 then it should at least in theory work for Joomla 5. Obviously there will be a point in the future where caution will once again be advised… As of writing we are at Joomla 5.0.1.

In a nut shell there are currently two points in the Joomla 4 + update cycle where you need to pay attention and avoid having a potential miss match of folders and or files. They are the updates: Joomla 4.1.0 and Joomla 4.4.4. In this Video Tutorial we are going to attempt to join the dots and figure out which Astroid Release make sense at what point in the Joomla Update cycle.

This and related tutorials are intended to be a nudge in the right direction. Speaking of related tutorials there will be a post that will dovetail into this in the future dealing with updating an existing install of Astroid where again the jumps in the Joomla update cycle mentioned above are something to keep an eye on.

**Reply by Chacapamac** [reply]
I think this, with the videos , is important as no real documentation is written on the subject and can easily cause discouragement for users. **Good Stuff Man!**

**Reply by sonvnn** [reply]
@paurray Great tutorial! I give you some notes that I hope are useful for your writing.

- If you are using Joomla 4 or Joomla 5. Let's use Astroid 3.x
- If you are stay with Joomla 3. Let's use Astroid 2.6.6

Thanks & Best Regards,
Sonny

**Comment by paurray** [comment]
@sonvnn @Chacapamac 
Since 3.0.9There is now a astroid-quickstart.zip
I guess that moving forward there will be a total 5 assets in the download area?

(I need to redo my work in progress tutorial to take this into account/Was almost finished!)

Can you please help me understand the assets?

astroid-framework-3.0.9.zip = Updating Manually? Correct?
astroid-package-v3.0.9.Framework-Template.zip = Fresh Install? Correct?
astroid-quickstart.zip = ???

thanks

**Reply by Chacapamac** [reply]
@paurray 

> astroid-framework-3.0.9.zip = Updating Manually? Correct?

— Yep

> astroid-package-v3.0.9.Framework-Template.zip = Fresh Install? Correct?

— Yes and No, You can use this to reinstall over an existing website fresh Astroid files for the Framework, and Templates.
The only point I’m not sure in this case is if somebody use the **Astroid Original Layouts** to build her/his website **_(something that nobody should do)_**. I’m not sure if it will erase the layouts customization. At 99%, I don’t think it will do anything.
—— For me, doing that is really useful if you think you have some kind of file corruption or a bad download.

> astroid-quickstart.zip = ???

— **This is new?**
**Can’t wait to try that**. Actually “Quick Start” are really cool for newbie or to developing rapidly a complete website. 

I think the goal here is to have a boilerplate website wit the basics pages and functions that are mostly always needed in a typical website.
— Cool Home page
— One About US or WhateverYouWant page
— One blog page/section
— One Contact page

— and some cool Astroid functions like, Page Banner and other stuff...

Something like that...

I’m actually working on a Template One “Quick Start” that highlights the coolest and trending Astroid abilities....
**COMING SOON — Hopefully!**

**Comment by paurray** [comment]
@Chacapamac 
thanks this helps :-)

**Reply by Chacapamac** [reply]
**Good Day Paul!**

**Comment by paurray** [comment]
@Chacapamac @sonvnn & Any One else who is following this
I finally got this tutorial finished and it is posted here:
https://astroidframe.work/blog/astroid-hands-on/how-to-get-up-and-running-the-traditional-way
The related video can also be found here on You Tube:
https://www.youtube.com/watch?v=CyXbOY7tQA0&t=1s

It ended up being quite different to how I planed it.

The whole introduction of  the “astroid-quickstart” asset really changed every thing.
I tried to make it as open as accessible to as many people as possible.
I had to leave out some details with respect to quirks eg around Joomla 4.1 or 4.4
Other wise things would have spiralled out of control = become really looooooooong
Could still be a "Lose Ends" section at a later date…

If any one has any specific constructive feedback back related to the article or video tutorial I would be very interested to hear from you.

Next planned move is “how-to-get-up-and-running-the-quickstart-way” or similar…

I will mark this thread as being solved in a couple of weeks.

thanks for all the help and support

**Reply by Chacapamac** [reply]
@paurray 
Watching it right now, seem to be pretty cool my friend.
**MY GENERAL IMPRESSION —THIS IS A FANTASTIC VIDEOS, GREAT WORK PAUL 🚀**
_(And, you know, at 64 and being a somewhat Autistic perfectionist, I’m really hard to impress)_👍

**QUESTIONS:** ( I put other questions as I’m watching now. 😀)
**1.** As minimum, don’t you have to assign the **default Joomla menu** to the Template to start to use Astroid?

**2.** For the Manual installation it will be cool if users know why this is sometimes used.
— e.g.  Resolver problems with the update systems or files corruption
—  Also you can, and maybe need in some cases, to use the package framework/template as it not replacing the user custom settings (only the files) — _(**I think I’m right here** Not totally sure, as I always use a custom layout to build a new website_. _(See Points Below...)_ 

**IMPORTANT POINTS**
• As a first install video, I think users should implement the minimum “Best Practice” and learn to keep the originals Astroid Template Styles unmodified and build their website on a custom Style. Anyway this is absolutely necessary if you make a multilingual website. It is also great to always have a “Clean Original Style” to generate new customs styles or as backup if problems. It is not clear on your video that this IS THE WAY TO GO.
— **For Sure, the best and pro way** is to use the “**Child Template**” Joomla/Astroid function, but for an easy startup video, I think it can be confusing for new users and Joomla newbie. — A complementary video should be added to this installation section showing that **Professional way to build Joomla/Astroid websites.**

**SUGGESTIONS**
• Maybe the Joomla Voting system should be on for these article to have some feedback (or some easy commenting system?

Again, relatively soon I will put online a “Quick Start” Template base on Template One, and ask the community to help me make it better.

**Comment by paurray** [comment, marked as answer]
Hello @Chacapamac 

Thanks for the positive feedback and thanks for pushing me in a very good fussy way :-) It really goes a long way. And I also have my defects. I hear you loud and clear.

With respect to 1: To the best of my knowledge Astroid just looks for what every is the set to the main Joomla Menu and takes that as it”s menu. Pretty much every thing I did in the Tutorial is unedited. So if you see it working in the tutorial that is how it should work. Please correct me if I am wrong.

With respect to 2: Yes of course you are right. This is a rabbit hole. I think it is a tough call. I tried to create a tutorial that makes Astroid as assessable to as many people as possible with as few potential red flags as possible. In it’s current form it runs at 13 min. The release of the Quickstart Asset really brought home to me the fact that the moment I upload a tutorial to You Tube it is out of date. In many ways it is an almost impossible task.

Patrick maybe something very very very roughly like this:

A Playlist “Cool Runnings with the Astroid Framework for Joomla”

With:

1) How to Get Up & Running the Traditional Way: DONE
2) How to Get Up & Running the Quick Start Way: To Do (Really need more help with this one)*
3) How to Maintain & Trouble shoot your Astroid Site: To Do (Really need more help with this one)*

*help with some of the more technical aspects eg how the whole quickstart thing will work in practice and/or editing child templates)

I think something to bear in mind is that the videos can reference each other.
So for example video (2) could skip the whole how Git Hub Releases are formatted thing and jump straight in with the Quickstart Asset! Or video (3) could refer to video (1) and (2) as you obviously have something up and running if you are trouble shooting/avoiding potential problems before they occur. Total estimated runtime for Playlist very roughly 20 -25 min!

- I think what I will do is start a new discussion and start gathering ideas for the next tutorial (2).

- Will close this thread here mid Feb.

- Also maybe a poll about how many people use child Templates. Kind of curious myself?!?

- “ Astroid Unleashed: How to Get Up and Running the Traditional Way”
https://youtu.be/CyXbOY7tQA0
The more engagement the better. Right?
